src/archive.rs

```diff
@@ -165,70 +196,73 @@ pub trait ArchiveBuilder: Clone {
 /// Macro used by generator to define a flatdata struct.
 #[macro_export]
 macro_rules! define_struct {
+
     // Simpler case where type and primitive_type coincide.
-    ($name:ident, $name_mut:ident, $schema:expr, $size_in_bytes:expr
+    ($factory:ident, $name:ident, $name_mut:ident, $schema:expr, $size_in_bytes:expr
         $(,($field:ident, $field_setter:ident, $type:tt, $offset:expr, $bit_size:expr))*) => {
-        define_struct!($name, $name_mut, $schema, $size_in_bytes
+        define_struct!($factory, $name, $name_mut, $schema, $size_in_bytes
             $(,($field, $field_setter, $type: $type, $offset, $bit_size))*
         );
     };
-    ($name:ident, $name_mut:ident, $schema:expr, $size_in_bytes:expr
+    ($factory:ident, $name:ident, $name_mut:ident, $schema:expr, $size_in_bytes:expr
         $(,($field:ident, $field_setter:ident, $type:tt: $primitive_type:tt, $offset:expr, $bit_size:expr))*) =>
     {
-        // TODO: We cannot store `&u8` here, since then we need to annotate the type with a
-        // lifetime, which would enforce an annotation in the trait, and this would bind the
-        // lifetime at the creating of containers as ArrayView, etc... When meta-types are
-        // introduced (i.e. when we can express that a container is parametrized over a meta-type
-        // with a lifetime bound later), we can refactor this and get rid of Handle and HandleMut.
         #[derive(Clone)]
-        pub struct $name {
+        pub struct $name<'a> {
             data: *const u8,
+            _phantom: $crate::marker::PhantomData<&'a u8>,
         }
 
-        impl $name {
+        pub struct $factory{}
+
+        impl<'a> $crate::Struct<'a> for $factory
+        {
+            const SCHEMA: &'static str = $schema;
+            const SIZE_IN_BYTES: usize = $size_in_bytes;
+
+            type Item = $name<'a>;
+            fn create(data : &'a[u8]) -> Self::Item
+            {
+                Self::Item{ data : data.as_ptr(), _phantom : $crate::marker::PhantomData }
```

Comment:
Do we actually need to store the pointer inside? Theoretically, since we are anyway binding the lifetime, we could just store the slice, right?

Reply:
Yes. I tried, but failed to make it nice: Basically we do not need any slice but a slice of a very specific (constant) size.

I have a few (less important) follow-up topics like that:
- Slice in type
- Checking that slice is big enough when constructing?
- Checking padding when constructing views

Reply:
Most likely using an array reference would work, plus some helpers that get arrays out of slices.

Reply:
Ok, lets defer it to a later point in time.